nilearn/_utils/tests/test_data_gen.py

```diff
@@ -15,6 +14,38 @@
 )
 from nilearn.image import get_data
 
+from nilearn._utils.data_gen import add_metadata_to_bids_dataset
+
+
+def test_add_metadata_to_bids_derivatives(tmp_path):
+    # bare bone smoke test
+    target_dir = tmp_path / 'derivatives' / 'sub-01' / 'ses-01' / 'func'
+    target_dir.mkdir(parents=True)
+    json_file = add_metadata_to_bids_dataset(bids_path=tmp_path,
+                                                  metadata={"foo": "bar"})
```

Comment:
I'm suprised by the code indentation here. What magic leads to that ?

Reply:
Until #3614 is in, the files under the _utils folder will not checked by flake8 so "technically valid albeit interesting" formats are to be expected.

We could wait for #3614 to be in before we merge this PR, or the other way around: I will fix those formatting you are flagging here, but just want to let you know that any "format drift" you see should be fixed soon.